arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -13,3 +13,18 @@ config ARM_MPU
 	default n
 	help
 	  MCU has ARM MPU
+
+config NXP_MPU
+	bool "NXP MPU Support"
+	depends on CPU_HAS_MPU
+	depends on SOC_FAMILY_KINETIS
+	default n
+	help
+	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
```

Comment:
CONFIG_FLASH doesn't imply that you want to write to flash.  It needs to be an additional option.